core/src/main/java/org/wso2/carbon/testgrid/core/TestGridUtil.java

```diff
@@ -39,17 +39,19 @@
     private static final Log log = LogFactory.getLog(TestGridUtil.class);
 
     static String createTestDirectory(String productName, String productVersion, Long timeStamp) throws IOException {
-        String directory = Paths.get(System.getenv(TESTGRID_HOME_ENV),productName + SEPARATOR + productVersion
+        String directory = Paths.get(System.getenv(TESTGRID_HOME_ENV), productName + SEPARATOR + productVersion
                 + SEPARATOR + timeStamp).toString();
-        File testDir = new File( directory);
+        File testDir = new File(directory);
         // if the directory exists, remove it
         if (testDir.exists()) {
             log.info("Removing test directory : " + testDir.getName());
             FileUtils.forceDelete(testDir);
         }
         log.info("Creating test directory : " + testDir.getName());
-        testDir.mkdir();
-        return directory;
+        if (testDir.mkdir()) {
+            return directory;
+        }
+        return null;
```

Comment:
Better to return an `Optional` rather than a `null` since there is a possibility of a null pointer exception when this method is called and if not handled by the developer.